keps/sig-apps/3715-elastic-indexed-job/README.md

```diff
@@ -0,0 +1,930 @@
+-->
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [x] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [x] (R) Design details are appropriately documented
+- [x] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [x] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Currently `spec.completions` is an immutable field in Job for both `Indexed` and `NonIndexed` modes. 
+This KEP proposes to allow mutating `spec.completions` for `Indexed` job iff the value in 
+the update is equal to `spec.parallelsim` specified in the update request. 
+
+Success and failure semantics are not changed for jobs that never mutate `spec.completions`; 
+however, for ones that do, the semantics are changed in the following manner:
+
+Failures will be counted against the lifetime of the job. If completions was scaled down, then previous pod failures
+outside the new range will still count against the job's backoffLimit. Morever, the field `status.Failed` will not be
+decremented if previously failed pods are now outside the range. However, `status.Succeeded` will be updated to reflect
+the number of successful indexes within the range as defined below.
+
+In the case where an index successfully completes, and then the job is scaled down rendering the index
+out of range, and then the job is scaled up again including back the index in the range, then the index 
+will be restarted.
+
```

Comment:
@soltysh @liggitt are we on agreement on the current semantics? The summary captures the latest proposal.

Reply:
lgtm

Reply:
lgtm as well

Reply:
Thanks Maciej and Jordan!